tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -3854,6 +3967,17 @@ <h2>Show similarity to selected datapoint</h2>
           }
         },
 
+        // Call backend for distance computation, backend calls callback function
+        // with computed distances and parameters
+        requestDistanceWithCallback: function(exInd, callbackFun, params) {
+          const urlParams = {
+            index: exInd,
+            callback: callbackFun,
+            params: params,
+          };
+          this.fire('compute-custom-distance', urlParams);
```

Comment:
I find it weird to see notebook specific code in the main dashboard code. What is the expected behavior outside of the notebook?

Reply:
requestDistanceWithCallback would not be invoked in non-local instances (local=demos and notebook). In case we are in non-local mode, WIT defaults to it's previous behavior of computing counterfactuals completely on the js side with L1 and L2 distance between examples. This is slightly similar in terms of behavior to custom_predict_fn where it is only supported in notebook mode.

Reply:
Added TODO note here so that if there is a support in TensorBoard mode to provide custom distance functions for counterfactuals then we should update this function to reflect that.